appinventor/appengine/src/com/google/appinventor/client/Ode.java

```diff
@@ -1429,6 +1285,39 @@ public void execute() {
       });
   }
 
+  /**
+   * Returns user new layout usage setting.
+   *
+   * @return true if the user has opted to use the new UI, false otherwise
+   */
+  public static boolean getUserNewLayout() {
+    String value = userSettings.getSettings(SettingsConstants.USER_GENERAL_SETTINGS).
+            getPropertyValue(SettingsConstants.USER_NEW_LAYOUT);
+    return Boolean.parseBoolean(value);
+  }
+
+  /**
+   * Set user new layout usage setting.
+   *
+   * @param newLayout new value for the user's UI preference
+   */
+  public static void setUserNewLayout(boolean newLayout) {
+    userSettings.getSettings(SettingsConstants.USER_GENERAL_SETTINGS).
+            changePropertyValue(SettingsConstants.USER_NEW_LAYOUT,
+                    "" + newLayout);
+    userSettings.saveSettings(new Command() {
+        @Override
+        public void execute() {
+          // Reload for the UI preferences to take effect. We
+          // do this here because we need to make sure that
+          // the user settings were saved before we terminate
+          // this browsing session. This is particularly important
+          // for Firefox
+          Window.Location.reload();
+        }
+      });
+  }
+
```

Comment:
We need to ultimately decide whether this ends up living in ODE. In the current UI binder code it isn't even used at all...

Reply:
> We need to ultimately decide whether this ends up living in ODE. In the current UI binder code it isn't even used at all...

I think it should probably go. I'll remove it. Jennet and I were working on user-selectable layouts, and I think the issues are thorny enough that we'll want to pull you in for some whiteboarding once the dust settles. The chance of these methods being of further use is slim.